Module.cs

```diff
@@ -86,30 +99,22 @@ protected override void Initialize()
         /// </summary>
         protected override async Task LoadAsync()
         {
-            await LoadSpecializationIconsAsync();
+            LoadSpecializationIcons();
             await LoadRoles();
         }
 
-        private async Task LoadSpecializationIconsAsync()
+        private void LoadSpecializationIcons()
         {
-            var connection = new Gw2Sharp.Connection();
-            using var client = new Gw2Sharp.Gw2Client(connection);
-            var webApiClient = client.WebApi.V2;
-
-            var professionsClient = webApiClient.Professions;
-            var professions = await professionsClient.AllAsync();
+            bool useTangoIcons = _areColorIconsEnabled.Value;
 
-            _professionIcons = professions.ToDictionary(
-                keySelector: (profession) => (uint)profession.Code,
-                (profession) => GameService.Content.GetRenderServiceTexture(profession.IconBig)
+            _professionIcons = Specialization.ProfessionCodes.ToDictionary(
+                keySelector: (profession) => profession,
+                (profession) => ContentsManager.GetTexture(Specialization.GetCoreIconPath(profession, useTangoIcons))
             );
 
-            var specializationClient = webApiClient.Specializations;
-            var eliteSpecs = await specializationClient.ManyAsync(Specialization.EliteCodes);
-
-            _specializationIcons = eliteSpecs.ToDictionary(
-                keySelector: (spec) => (uint)spec.Id,
-                (spec) => GameService.Content.GetRenderServiceTexture(spec.ProfessionIconBig)
+            _specializationIcons = Specialization.EliteCodes.ToDictionary(
+                keySelector: (spec) => (uint)spec,
```

Comment:
I think the cast can be removed, as spec is already `uint`

Reply:
Had to change the type of `IReadOnlyCollection<int> EliteCodes` to `IReadOnlyCollection<uint> EliteCodes` to get rid of the cast